Userland/Libraries/LibPDF/Document.cpp

```diff
@@ -108,16 +108,21 @@ PDFErrorOr<Page> Document::get_page(u32 index)
     auto page_object_index = m_page_object_indices[index];
     auto page_object = TRY(get_or_load_value(page_object_index));
     auto raw_page_object = TRY(resolve_to<DictObject>(page_object));
+    auto parent = TRY(raw_page_object->get_dict(this, CommonNames::Parent));
 
-    if (!raw_page_object->contains(CommonNames::Resources)) {
-        // This page inherits its resource dictionary
-        TODO();
-    }
+    auto get_inheritable_object = [&](FlyString const& name) -> PDFErrorOr<NonnullRefPtr<Object>> {
```

Comment:
Can the inheritance chain be longer than 2? I.e., A inherits from B and B inherits from C. If so, we might want to pull this into a standalone function so it can be called recursively.

Reply:
I'm not quite sure from reading the spec, we would probably need a test document that depends on it.

Reply:
Reading the 1.7 spec here: https://opensource.adobe.com/dc-acrobat-sdk-docs/standards/pdfstandards/pdf/PDF32000_2008.pdf

```
7.7.3.4 Inheritance of Page Attributes

Some of the page attributes shown in Table 30 are designated as inheritable. If such an attribute is omitted from
a page object, its value shall be inherited from an ancestor node in the page tree. If the attribute is a required
one, a value shall be supplied in an ancestor node. If the attribute is optional and no inherited value is specified,
the default value shall be used.

An attribute can thus be defined once for a whole set of pages by specifying it in an intermediate page tree
node and arranging the pages that share the attribute as descendants of that node.

EXAMPLE A document may specify the same media box for all of its pages by including a MediaBox entry in the root
node of the page tree. If necessary, an individual page object may override this inherited value with a
MediaBox entry of its own.
```

Both MediaBox and Resources are in Table 30 in that spec document, which is telling us that yes, we must walk up the entire ancestor tree of a page looking for the attribute before going to a fallback.

Reply:
Right, that's fixed now. Sorry for that :^)